Build failure: kitty-0.42.2 broken by #446758

### Nixpkgs version

- nixpkgs/master

### Steps to reproduce

nixos-rebuild with nixpkgs/master

### Can Hydra reproduce this build failure?

Please select the Hydra Status.

### Link to Hydra build job

_No response_

### Relevant log output

```console
building '/nix/store/7rxlrs63rm5ln66swa6c55rif0lfnw9f-fish-4.1.0.drv'...
building '/nix/store/25wxpgbdrygxck2r7pfpmwdbi9sqagh1-kitty-0.42.2.drv'...
note: keeping build directory '/tmp/nix-build-kitty-0.42.2.drv-0/build'
error: builder for '/nix/store/25wxpgbdrygxck2r7pfpmwdbi9sqagh1-kitty-0.42.2.drv' failed with exit code 2;
       last 25 log lines:
       > FAIL: test_fish_integration (kitty_tests.shell_integration.ShellIntegrationWithKitten.test_fish_integration)
       > ----------------------------------------------------------------------
       > Traceback (most recent call last):
       >   File "/build/source/kitty/launcher/../../kitty_tests/shell_integration.py", line 203, in test_fish_integration
       >     self.ae(pty.screen_contents(), q)
       >     ~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^
       >     completions_dir = '/build/source/shell-integration/fish/vendor_completions.d'
       >     fish_prompt = 'left>'
       >     pty = <kitty_tests.PTY object at 0x7fffe4124cb0>
       >     q = 'left>                                                                                                                                                                         <right'
       >     right_prompt = '<right'
       >     self = <kitty_tests.shell_integration.ShellIntegrationWithKitten testMethod=test_fish_integration>
       > AssertionError: "warning: fish could not read response to[471 chars]ight" != 'left>                                   [136 chars]ight'
       > - warning: fish could not read response to primary device attribute query after waiting for 2 seconds. This is often due to a missing feature in your terminal. See 'help terminal-com
       > - patibility' or 'man fish-terminal-compatibility'. This fish process will no longer wait for outstanding queries, which disables some optional features.
       >   left>                                                                                                                                                                         <right
       >
       >
       > ----------------------------------------------------------------------
       > Ran 156 tests in 11.423s
       >
       > FAILED (failures=2, skipped=2)
       > All Go tests succeeded, ran in 11.6 seconds
       > Error: Some tests failed!
       > make: *** [Makefile:16: test] Error 1
       For full logs, run:
         nix log /nix/store/25wxpgbdrygxck2r7pfpmwdbi9sqagh1-kitty-0.42.2.drv
```

### Additional context

"[adamcstephens](https://github.com/adamcstephens): I skipped building kitty and mise because they are large builds and are failing. If indeed this two apps break, they will need to be fixed by those teams anyway.

fishPlugins.fzf-fish is already failing on darwin."
https://github.com/NixOS/nixpkgs/pull/446758#issuecomment-3342062817

### System metadata

❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.16.9, NixOS, 25.11 (Xantusia), 25.11.20250928.6f6889a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.5`
 - nixpkgs: `/nix/store/r0n7qmp6hxg1r7nnhnq6zmsvckk50iz7-source`

### Notify maintainers

@tex @rvolosatovs @Luflosi @kashw2 @leiserfg

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ki/kitty/package.nix
+++ b/pkgs/by-name/ki/kitty/package.nix
@@ -45,26 +45,27 @@
   makeBinaryWrapper,
   autoSignDarwinBinariesHook,
   cairo,
+  fetchpatch,
 }:
 
 with python3Packages;
 buildPythonApplication rec {
   pname = "kitty";
-  version = "0.42.2";
+  version = "0.43.0";
   format = "other";
 
   src = fetchFromGitHub {
     owner = "kovidgoyal";
     repo = "kitty";
     tag = "v${version}";
-    hash = "sha256-YDfKYzj5LRx1XaKUpBKo97CMW4jPhVQq0aXx/Qfcdzo=";
+    hash = "sha256-wPLXuZWhaA51J7jGHffh/xnXzWDKCXV2G3Uvrg7G8Kg=";
   };
 
   goModules =
     (buildGo124Module {
       pname = "kitty-go-modules";
       inherit src version;
-      vendorHash = "sha256-q5LMyogAqgUFfln7LVkhuXzYSMuYmOif5sj15KkOjB4=";
+      vendorHash = "sha256-bjtzvEQmpsrwD0BArw9N6/HqMB3T5xeqxpx89FV7p2A=";
     }).goModules;
 
   buildInputs = [
@@ -142,6 +143,12 @@ buildPythonApplication rec {
     # OSError: master_fd is in error condition
     ./disable-test_ssh_bootstrap_with_different_launchers.patch
 
+    # Fix test failure with fish >= 4.1
+    # See: https://github.com/kovidgoyal/kitty/commit/2f991691f9dca291c52bd619c800d3c2f3eb0d66
+    (fetchpatch {
+      url = "https://github.com/kovidgoyal/kitty/commit/2f991691f9dca291c52bd619c800d3c2f3eb0d66.patch";
+      hash = "sha256-LIQz3e2qgiwpsMd5EbEcvd7ePEEPJvIH4NmNpxydQiU=";
+    })
   ];
 
   hardeningDisable = [

